app/src/main/java/com/lifespandh/ireflexions/home/care/CareCenterFragment.kt

```diff
@@ -31,24 +48,27 @@ class CareCenterFragment : BaseFragment() {
     }
 
     private fun init() {
+        homeViewModel.getSupportContacts()
+        setViews()
         setListeners()
+        setObservers()
+    }
+
+    private fun setViews() {
+        support_contacts_recyclerView.apply {
+            adapter = supportContactAdapter
+            layoutManager = LinearLayoutManager(requireContext())
+        }
     }
 
     private fun setListeners() {
         addContactCardView.setOnClickListener {
             if (sharedPrefs.isLoggedIn) {
                 if (requireActivity().checkSelfPermission(android.Manifest.permission.READ_CONTACTS) == PackageManager.PERMISSION_GRANTED) {
-                    findNavController().navigate(R.id.editSupportContactFragment)
+                    val action = CareCenterFragmentDirections.actionCareCenterFragmentToEditSupportContactFragment(null, true)
+                    findNavController().navigate(action)
                 } else {
-                    PermissionLauncher(this, object : PermissionLauncher.OnPermissionResult {
-                        override fun onPermissionGranted() {
-
-                        }
-
-                        override fun onPermissionDenied() {
-
-                        }
-                    }).launch(android.Manifest.permission.READ_CONTACTS)
+                    permissionLauncher.launch(android.Manifest.permission.READ_CONTACTS)
                 }
             } else {
                 showDialog(
```

Comment:
@RudraNagalia change the message here.
Just add new string resources and change the message. It says cannot show courses and lessons without login, but it should show cannot add new contact without login ....., in case of care center